src/cls/rbd/cls_rbd.cc

```diff
@@ -5107,6 +5107,86 @@ int mirror_image_map_remove(cls_method_context_t hctx, bufferlist *in,
   return 0;
 }
 
+/*********************** methods for rbd_group_directory ***********************/
+
+static int group_dir_add_helper(cls_method_context_t hctx,
```

Comment:
Nit: no need for ```static``` nor ```group_``` prefix when you put these helper methods in the existing ```group``` namespace. Also, the suffix of ```helper``` is not very descriptive as to what the purpose of this function is -- I think you can just use ```int dir_add``` within the namespace.